spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraStrategies.scala

```diff
@@ -121,7 +121,8 @@ private[cassandra] trait CassandraStrategies {
         val eqPredicates = eqColumns.flatMap(eqPredicatesByName)
         val optionalNonEqPredicate = for {
           c <- otherColumns.headOption.toSeq
-          p <- firstNonEmptySeq(rangePredicatesByName(c), inPredicatesByName(c).filter(_ => eqColumns.nonEmpty))
+          p <- firstNonEmptySeq(rangePredicatesByName(c), inPredicatesByName(c).filter(
+            _ => eqColumns.nonEmpty && c==clusteringColumns.last))
```

Comment:
I remove eqColumns.nonEmpty  condition. We only need c==clusteringColumns.last condition, so that only if the last cluster column with IN clause can be pushed down.